src/rust/util/src/lib.rs

```diff
@@ -50,6 +50,25 @@ impl<'a, T> Survive<'a, T> {
     }
 }
 
+/// A wrapper that adds the Sync trait to RefCell. We can use this to avoid mutexes as our target is
+/// single-threaded and our unit tests run single-threaded.
```

Comment:
> our target is single-threaded

Our target isn't single-threaded, it has interrupts. So the comment should say more something in line with the fact that we don't want to fix this right now, that we are breaking the rust soundness, and that we intend to fix it later. Maybe put a "TODO" next to the `unsafe impl Sync`.

Reply:
Hmm good point. For now I moved it to testing.rs as it's only used in unit tests. Is the assumption there okay as we run single-threaded and there are no interrupts there?

Btw, we use the same trick also in production code since a long time (probably incorrect):
https://github.com/BitBoxSwiss/bitbox02-firmware/blob/160e154d8cae9211feb26e72f85bb84f4c5be70c/src/rust/bitbox02-rust/src/async_usb.rs#L44-L49
 
Though I have never seen a panic in the use of this particular RefCell myself or from a customer. 

Possibly useful reference: https://docs.rust-embedded.org/book/concurrency/

Reply:
you also have this: https://doc.rust-lang.org/nomicon/send-and-sync.html